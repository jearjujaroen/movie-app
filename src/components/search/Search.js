import React from 'react';
import PropTypes from 'prop-types';
import styles from './Search.module.css';

export const Search = ({ handleInputChange }) => (
  //TODO: implement debounce or use Lodash debounce when searching
  <div className={styles.searchBar}>
    <form>
      <label htmlFor='search'></label>
      <input
        placeholder='Search'
        name='search'
        onChange={e => handleInputChange(e)}
      />
      <svg
        xmlns='http://www.w3.org/2000/svg'
        viewBox='0 0 12 13'
        width='20'
        height='21'
      >
        <g strokeWidth='2' stroke='#6c6c6c' fill='none'>
          <path d='M11.29 11.71l-4-4' />
          <circle cx='5' cy='5' r='4' />
        </g>
      </svg>
    </form>
  </div>
);

Search.propTypes = {
  handleInputChange: PropTypes.func.isRequired,
};
