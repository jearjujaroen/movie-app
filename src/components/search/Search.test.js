import React from 'react';
import renderer from 'react-test-renderer';
import { Search } from './Search';

describe('testing search', () => {
  it('renders correctly', () => {
    const handleInputChange = e => {
      console.log(e);
    };

    const item = renderer
      .create(<Search handleInputChange={handleInputChange} />)
      .toJSON();
    expect(item).toMatchSnapshot();
  });
});
