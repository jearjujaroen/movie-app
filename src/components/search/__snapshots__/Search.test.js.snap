// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing search renders correctly 1`] = `
<div
  className="searchBar"
>
  <form>
    <label
      htmlFor="search"
    />
    <input
      name="search"
      onChange={[Function]}
      placeholder="Search"
    />
    <svg
      height="21"
      viewBox="0 0 12 13"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        stroke="#6c6c6c"
        strokeWidth="2"
      >
        <path
          d="M11.29 11.71l-4-4"
        />
        <circle
          cx="5"
          cy="5"
          r="4"
        />
      </g>
    </svg>
  </form>
</div>
`;
